flask/app/groups.py

```diff
@@ -99,7 +108,11 @@ def update_group(group_code):
     )
 
     # Check if a user to be added doesn't exist, 404
-    if "users" in request.json:
+    if (
+        "users" in request.json
+        and type(request.json["users"]) is list
+        and all([type(user) is str for user in request.json["users"]])
+    ):
```

Comment:
the group name update occurs regardless of whether this check succeeds or not - is this intended behaviour?

Reply:
This is intended. The `users` field is optional. If not provided, no change is made.

Reply:
what i meant (similar to below) is that if `users` is supplied but not in the correct format (eg. not in a list) then the group name update will still occur and there is no indication that the supplied `users` field was invalid. i don't think it's an issue, i'm just wondering if it would be helpful to at least log some kind of message indicating there was an attempt to do so

Reply:
I've accounted for this now too. Maybe another time we can look at the input validation problem in depth.

Reply:
nice - i agree, it would also help in reducing repetitive code across the endpoints